docs/src/docs/hooks/use-validated-state.mdx

```diff
@@ -0,0 +1,49 @@
+---
+group: 'mantine-hooks'
+package: '@mantine/hooks'
+title: 'use-validated-state'
+category: 'state'
+order: 1
+slug: /hooks/use-validated-state/
+description: 'useValidatedState is useState with validation'
+import: "import { useValidatedState } from '@mantine/hooks';"
+docs: 'hooks/use-validated-state.mdx'
+source: 'mantine-hooks/src/use-validated-state/use-validated-state.ts'
+---
+
+## Usage
+
+use-validated-state works similar to `useState` but checks the value if it is valid based on the `rule` function.
```

Comment:
Change description to:
"Hook validates state with a given rule each time state is set. It returns an object with current validation state, last valid value and current value:"